### PR TITLE
feat: enable multiquery

### PR DESCRIPTION
**Pull request by ariadnafer**
# Description

Previous custom components was not allowing running multi-queries in the same SQL file. This PR merges all the queries into a single SQL script and replace the custom component with the BigqueryQueryJobOp google component

# How has this been tested?

Please explain how you have tested the new changes.

# Checklist

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have successfully run the E2E tests, and have included the links to the pipeline runs below
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have updated any relevant documentation to reflect my changes
- [x] I have assigned a reviewer and messaged them

# Pipeline run links:

https://console.cloud.google.com/vertex-ai/locations/europe-west2/pipelines/runs/xgboost-train-pipeline-20230516134111?project=dt-turbo-templates-dev
https://console.cloud.google.com/vertex-ai/locations/europe-west2/pipelines/runs/xgboost-prediction-pipeline-20230516135109?project=dt-turbo-templates-dev
https://console.cloud.google.com/vertex-ai/locations/europe-west2/pipelines/runs/tensorflow-train-pipeline-20230516135752?project=dt-turbo-templates-dev
https://console.cloud.google.com/vertex-ai/locations/europe-west2/pipelines/runs/tensorflow-prediction-pipeline-20230516142755?project=dt-turbo-templates-dev

### CODE REVIEW

File: pipelines/src/pipelines/xgboost/training/queries/ingest.sql [hunk 1]
```diff
@@ -1,39 +1,32 @@
--- Copyright 2022 Google LLC
+CREATE SCHEMA IF NOT EXISTS `{{ preprocessing_dataset }}`
```
Comment (browningjp-datatonic) [inline pipelines/src/pipelines/xgboost/training/queries/ingest.sql:1]: likewise here please could you add some more comments

File: pipelines/src/pipelines/__init__.py [hunk 2]
```diff
@@ -30,4 +30,10 @@ def generate_query(input_file: Path, **replacements) -> str:
     with open(input_file, "r") as f:
         query_template = f.read()
 
-    return Template(query_template).render(**replacements)
+    # Render the template with the provided replacements
+    query = Template(query_template).render(**replacements)
+
+    # Escape double quotes, newline and tab characters
+    query = query.replace('"', '\\"').replace("\n", "\\n").replace("\t", "\\t")
```
Comment (browningjp-datatonic) [inline pipelines/src/pipelines/__init__.py:37]: is the first one correct? there are two `\`

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by felix-datatonic — DISMISSED** [review]

File: pipelines/src/pipelines/__init__.py (hunk 2 above)
Comment (ariadnafer) [reply pipelines/src/pipelines/__init__.py:37]: Well, the reason behind this new line is:

- In the old component, the query that we needed to pass was a simple string, where we already were generated \n and \t characters.

- In the new component, this string live inside a JSON, and for loading the query executes a `json.loads` command that fails in interpreting the escaping commands for some reason I don't fully understand. So, if we want to preserve the exact format of the string, we need to escape also the special characters.

Comment (browningjp-datatonic) [reply pipelines/src/pipelines/__init__.py:37]: Is it just the double-quotes issue? What happens if we only use single quotes `'` in the SQL script?

Comment (ariadnafer) [reply pipelines/src/pipelines/__init__.py]: mmm it seems it worked :) `json.loads` was loading properly the line breaks but not the quotes. Sorry I forgot to check that!

File: pipelines/src/pipelines/tensorflow/prediction/queries/ingest.sql [hunk 3]
```diff
@@ -14,8 +14,16 @@
 
 -- Treat "filter_start_value" as the current time, unless it is empty then use CURRENT_DATETIME() instead
 -- This allows us to set the filter_start_value to a specific time for testing or for backfill
+CREATE SCHEMA IF NOT EXISTS `{{ prediction_dataset }}`
```
[the remark this answers is not in the document]
Comment (ariadnafer) [reply pipelines/src/pipelines/tensorflow/prediction/queries/ingest.sql:19]: changed :)

File: pipelines/src/pipelines/__init__.py
Comment (browningjp-datatonic) [inline pipelines/src/pipelines/__init__.py:1]: @ariadnafer please could you revert the changes in this file? Looks like they are no longer needed?

File: pipelines/src/pipelines/tensorflow/prediction/pipeline.py [hunk 4]
```diff
@@ -88,23 +89,20 @@ def tensorflow_pipeline(
     queries_folder = pathlib.Path(__file__).parent / "queries"
 
     ingest_query = generate_query(
```
Comment (browningjp-datatonic) [inline pipelines/src/pipelines/tensorflow/prediction/pipeline.py:91]: maybe rename `ingest_query` and the `ingest` task to `preprocessing_query` and `preprocessing`

**Review by ghost — CHANGES_REQUESTED** [review]

File: pipelines/src/pipelines/tensorflow/prediction/pipeline.py (hunk 4 above)
Comment (ariadnafer) [reply pipelines/src/pipelines/tensorflow/prediction/pipeline.py]: done

File: pipelines/src/pipelines/__init__.py
Comment (ariadnafer) [reply pipelines/src/pipelines/__init__.py:1]: done. It was already changed but with some different formatting

File: pipelines/src/pipelines/tensorflow/training/pipeline.py [hunk 5]
```diff
@@ -97,90 +97,38 @@ def tensorflow_pipeline(
...
-        destination_project_id=project_id,
-        dataset_id=dataset_id,
-        dataset_location=dataset_location,
-        query_job_config=json.dumps(dict(write_disposition="WRITE_TRUNCATE")),
-    )
-    ingest = bq_query_to_table(
-        query=ingest_query, table_id=ingested_table, **kwargs
+    preprocessing = BigqueryQueryJobOp(
+        project=project_id, location=dataset_location, query=preprocessing_query
     ).set_display_name("Ingest data")
```
Comment (felix-datatonic) [inline pipelines/src/pipelines/tensorflow/training/pipeline.py:117]:
```suggestion
    ).set_display_name("Ingest & preprocess data")
```

File: pipelines/src/pipelines/xgboost/training/pipeline.py [hunk 6]
```diff
@@ -95,89 +95,38 @@ def xgboost_pipeline(
...
-        destination_project_id=project_id,
-        dataset_id=dataset_id,
-        dataset_location=dataset_location,
-        query_job_config=json.dumps(dict(write_disposition="WRITE_TRUNCATE")),
-    )
-    ingest = bq_query_to_table(
-        query=ingest_query, table_id=ingested_table, **kwargs
+    preprocessing = BigqueryQueryJobOp(
+        project=project_id, location=dataset_location, query=preprocessing_query
     ).set_display_name("Ingest data")
```
Comment (felix-datatonic) [inline pipelines/src/pipelines/xgboost/training/pipeline.py:115]:
```suggestion
    ).set_display_name("Ingest & preprocess data")
```

**Review by felix-datatonic — APPROVED** [review]

**Review by ghost — APPROVED** [review]
LGTM!
